sklearn/metrics/ranking.py

```diff
@@ -719,3 +720,67 @@ def label_ranking_loss(y_true, y_score, sample_weight=None):
     loss[np.logical_or(n_positives == 0, n_positives == n_labels)] = 0.
 
     return np.average(loss, weights=sample_weight)
+
+
+def precision_at_k_score(y_true, y_score, sample_weight=None, n_tops=1):
+    """Compute the precision for n_tops scored labels
+
+    Count how many time the ``n_tops`` label with highest score are in the set
+    of true labels. One minus precision at 1 is also called one error.
+    It's averaged over the samples and can take a value between 0 and 1.
+    The best performance achievable could be below 1.
+
+    Parameters
+    ----------
+    y_true : array, shape = [n_samples, n_labels]
+        True binary labels in binary indicator format.
+
+    y_score : array, shape = [n_samples, n_labels]
+        Target scores, can either be probability estimates of the positive
+        class, confidence values, or binary decisions.
+
+    sample_weight : array-like of shape = [n_samples], optional
+        Sample weights.
+
+    n_tops : int, optional
+        The number of labels on which the precision is computed.
```

Comment:
In ivalice, I am using `max_rank` for this. I think I got the name from the LambdaMART PR.